core/services/synchronization/telemetry_ingress_batch_client.go

```diff
@@ -182,16 +172,23 @@ func (tc *telemetryIngressBatchClient) getCSAPrivateKey() (privkey []byte, err e
 // Send directs incoming telmetry messages to the worker responsible for pushing it to
 // the ingress server. If the worker telemetry buffer is full, messages are dropped
 // and a warning is logged.
-func (tc *telemetryIngressBatchClient) Send(payload TelemPayload) {
+func (tc *telemetryIngressBatchClient) Send(ctx context.Context, telemData []byte, contractID string, telemType TelemetryType) {
 	if tc.useUniConn && !tc.connected.Load() {
 		tc.lggr.Warnw("not connected to telemetry endpoint", "endpoint", tc.url.String())
 		return
 	}
+	payload := TelemPayload{
+		Ctx:        ctx,
```

Comment:
@george-dorin This isn't right: the sending process (worker) should use its own context (`ctx.Background()` I assume, since it's working async) once the message has been delegated to it, but this context until then (so using ctx.Done on l 191 is correct).

Reply:
Fixed in [`aeb4595` (#10623)](https://github.com/smartcontractkit/chainlink/pull/10623/commits/aeb459564ca8d8601996980547446b7e8232f4d4)